pkg/configv2/types/bacalhau.go

```diff
@@ -0,0 +1,60 @@
+package types
+
+// Bacalhau represents the configuration for the Bacalhau system,
+// including client, server, orchestrator, compute service, and telemetry settings.
+type Bacalhau struct {
+	// Repo specifies a path on the filesystem where Bacalhau will persist its state.
+	Repo string `yaml:"Repo,omitempty"`
+	// Name specifies the name the Bacalhau node will advertise on the network.
+	Name string `yaml:"Name,omitempty"`
+	// Client specifies the configuration of the Bacalhau client.
+	Client Client `yaml:"Client,omitempty"`
+	// Server specifies the configuration of the Bacalhau server.
+	Server Server `yaml:"Server,omitempty"`
```

Comment:
lets just have `API` that represents both how client can connect to nodes, and how nodes can listen. This is easier for users to understand how to setup things up and how they relate to each other. If configs conflict somehow, then they can define a `node.yaml` config and `client.yaml` config. 

Besides, isn't the plane to have client configs separate using contexts or we don't think that is going to be feasible for 1.5?

Reply:
The `Server` has configuration options for TLS and Auth which the `Client` doesn't have so I'd like to keep these things separate.

Reply:
Separating them in our existing config is causing a lot of confusion. TLS should apply to both, even if that is not the case today. Also it should be fine if _some_ fields are client or server specific as the majority of other fields are the same. Lately, users shouldn't use the same config file for both running a node and calling client operations. We are making things easier by allowing them to easily define their config file, and the default config paths are (should be) different as we discussed in our last 1:1